### PR TITLE
🏗️ Build spider: Atlantic City

### PR DESCRIPTION
**Pull request by msrezaie**
## What's this PR do?

Adds a spider to scrape meeting information of Atlantic City. The scraper uses API endpoints to fetch the meeting data.

## Why are we doing this?

Scraper requested from [spreadsheet](https://docs.google.com/spreadsheets/d/1-HwPQKwm56vZ8h7CEM65GazKLTS0086gmaEzEHXp7O4/edit?usp=sharing).

## Steps to manually test
1. Ensure the project is installed:
```
pipenv sync --dev
```
2. Activate the virtual env and enter the pipenv shell:
```
pipenv shell
```
3. Run the spider:
```
scrapy crawl atconj_Atlantic_City -O test_output.csv
```
4. Monitor the output and ensure no errors are raised.
5. Inspect `test_output.csv` to ensure the data looks valid.
6. Ensure all tests pass
```
pytest
```

## Are there any smells or added technical debt to note?
No

## Summary by CodeRabbit

- **New Features**
	- Introduced a web scraper for Atlantic City meetings, enabling users to access meeting data from a dedicated API.
	- Added JSON files containing meeting entries and detailed meeting information for enhanced data management.

- **Tests**
	- Implemented a suite of unit tests to validate the functionality of the Atlantic City scraper, ensuring accurate data parsing and integrity.

### CODE REVIEW

File: city_scrapers/spiders/atconj_Atlantic_City.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    - `meetings_url`: provides a list of all meetings for a given time
+    period.
+    - `meeting_detail_url`: retrieves detailed information for each
+    meeting using its ID.
+
+    Additionally, a third url `calendar_source` is used as the source
+    field of the meeting since it is more user friendly to navigate
+    than the api endpoints.
+    """
+    meetings_url = "https://www.acnj.gov/api/data/GetCalendarMeetings?end=06%2F30%2F2025+12:00+am&meetingTypeID=all&start=06%2F01%2F2024+12:00+am"  # noqa
```
Comment (lamle-ea) [inline city_scrapers/spiders/atconj_Atlantic_City.py]: Please ensure that the values of the query parameters `start` and `end` in the meeting URL are dynamic and based on the current date.

File: city_scrapers/spiders/atconj_Atlantic_City.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                "name": "City Hall of Atlantic City",
+            }
+
+    def _parse_links(self, item):
+        base_url = "https://www.acnj.gov/"
+        keys = ["Meeting_AgendaPDF", "Meeting_MinutesPDF", "Meeting_NoticePDF"]
+        titles = ["Agenda", "Minutes", "Notice"]
+
+        links = [
+            {"title": title, "href": urljoin(base_url, item.get(key, ""))}
```
Comment (lamle-ea) [inline city_scrapers/spiders/atconj_Atlantic_City.py]: If you have already checked the condition `item.get(key)`, there’s no need to include the fallback value `""` here.

File: city_scrapers/spiders/atconj_Atlantic_City.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    """
+    meetings_url = "https://www.acnj.gov/api/data/GetCalendarMeetings?end=06%2F30%2F2025+12:00+am&meetingTypeID=all&start=06%2F01%2F2024+12:00+am"  # noqa
+    meeting_detail_url = "https://www.acnj.gov/api/data/GetMeeting?id="
+    calendar_source = "https://www.acnj.gov/calendar"
+
+    def start_requests(self):
+        yield scrapy.Request(url=self.meetings_url, method="GET", callback=self.parse)
+
+    def parse(self, response):
+        data = json.loads(response.text)
```
Comment (cruznunez) [inline city_scrapers/spiders/atconj_Atlantic_City.py]: try this
```suggestion
        data = response.json()
```

**Review by cruznunez — CHANGES_REQUESTED** [review]
besides the issue with the hardcoded date, we can also remove the json dependency.

File: city_scrapers/spiders/atconj_Atlantic_City.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
+import json
```
Comment (cruznunez) [inline city_scrapers/spiders/atconj_Atlantic_City.py]:
```suggestion
```

**Review by cruznunez — CHANGES_REQUESTED** [review]
Nice work with `relativedelta`. That's pretty cool. 1 more change with json dependency and we should be good.

File: city_scrapers/spiders/atconj_Atlantic_City.py (hunk 4 above)
Comment (msrezaie) [reply city_scrapers/spiders/atconj_Atlantic_City.py]: Good catch. I was still using `json.loads` on the meeting details method. That is now refactored. Thanks!

**Review by cruznunez — APPROVED** [review]
LGTM

**Review by lamle-ea — APPROVED** [review]
LGTM

**Pull request merged**